test/optimization/test_quadratic_program.py

```diff
@@ -130,27 +128,111 @@ def assert_equal(x: Variable, y: Variable):
-            q_p.var_dict(name='c0')
+            q_p.binary_var_dict(name='c0')
+
+        with self.assertRaises(QiskitOptimizationError):
+            q_p.binary_var_dict(name='c', keys=range(3))
+
+        with self.assertRaises(QiskitOptimizationError):
+            q_p = QuadraticProgram()
+            q_p.binary_var_dict(formatter='{}{}')
+
+        with self.assertRaises(QiskitOptimizationError):
+            q_p = QuadraticProgram()
+            q_p.binary_var_dict(keys=0)
+
+        with self.assertRaises(QiskitOptimizationError):
+            q_p = QuadraticProgram()
+            q_p.binary_var_dict(name='a')
+            q_p.binary_var_dict(name='a')
+
+        with self.assertRaises(QiskitOptimizationError):
+            q_p = QuadraticProgram()
+            q_p.binary_var_dict(formatter='_{{}}')
+
+    def test_var_list(self):
+        """test {binary,integer,continuous}_var_list"""
+        q_p = QuadraticProgram()
+
+        c_count = 0
+        b_count = 0
+        i_count = 0
+
+        def verify_counts():
+            self.assertEqual(q_p.get_num_vars(), c_count + b_count + i_count)
+            self.assertEqual(q_p.get_num_continuous_vars(), c_count)
+            self.assertEqual(q_p.get_num_binary_vars(), b_count)
+            self.assertEqual(q_p.get_num_integer_vars(), i_count)
+
+        def check_dict(var_dict, offset):
+            verify_counts()
+            variables = var_dict.values()
+            for i, x in enumerate(variables):
+                y = q_p.get_variable(i + offset)
+                z = q_p.get_variable(x.name)
+                assert_equal(x, y)
+                assert_equal(x, z)
+
+        def check_list(var_list, offset):
+            verify_counts()
+            for i, x in enumerate(var_list):
+                y = q_p.get_variable(i + offset)
+                z = q_p.get_variable(x.name)
+                assert_equal(x, y)
+                assert_equal(x, z)
+
+        def assert_equal(x: Variable, y: Variable):
+            self.assertEqual(x.name, y.name)
+            self.assertEqual(x.lowerbound, y.lowerbound)
+            self.assertEqual(x.upperbound, y.upperbound)
+            self.assertEqual(x.vartype, y.vartype)
+
+        d_0 = q_p.continuous_var_list(name='a', formatter='_{}', keys=3)
+        c_count += 3
+        check_list(d_0, 0)
+
+        d_1 = q_p.binary_var_list(name='b', keys=5)
+        b_count += 5
+        check_list(d_1, len(d_0))
+
+        d_2 = q_p.integer_var_list(formatter='_{}', keys=7, lowerbound=-4, upperbound=10)
+        i_count += 7
+        check_list(d_2, len(d_0) + len(d_1))
+
+        d_3 = q_p.continuous_var_list(name='a', formatter='_{}', keys=3)
+        c_count += 3
+        check_list(d_3, len(d_0) + len(d_1) + len(d_2))
+
+        d_4 = q_p.continuous_var_list(name='c', keys=range(3))
+        c_count += 3
+        check_list(d_4, len(d_0) + len(d_1) + len(d_2) + len(d_3))
+
+        d_5 = q_p.binary_var_dict(name='d', keys=range(5))
```

Comment:
Now that the list test cases have been expanded, I don't think this test is necessary. The same goes for the `binary_var_list` use in `test_var_dict`.